### PR TITLE
timers: Replace DEBUGASSERT with the error code

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

fix the issue report here:
https://lists.apache.org/thread/sys37yf63rq501fd1v8y3zyh6vk10v1d

## Impact

more robust

## Testing

ci

### CODE REVIEW

File: include/nuttx/timers/oneshot.h [hunk 1]
```diff
@@ -266,7 +266,10 @@ int oneshot_max_delay(FAR struct oneshot_lowerhalf_s *lower,
   clock_t tick;
   int ret;
 
-  DEBUGASSERT(lower->ops->tick_max_delay);
+  if (lower->ops->tick_max_delay == NULL)
+    {
+      return -ENOSYS;
```
Comment (patacongo) [inline include/nuttx/timers/oneshot.h:271]: In a traditional Unix system, ENOSYS is returned only if an entire system call is not supported.  ENOSUPP is the correct error return if a feature or function of a system call is not supported.

Comment (xiaoxiang781216) [reply include/nuttx/timers/oneshot.h]: Done

File: include/nuttx/timers/timer.h [hunk 2]
```diff
@@ -253,7 +253,10 @@ int timer_getstatus(FAR struct timer_lowerhalf_s *lower,
 {
   int ret;
 
-  DEBUGASSERT(lower->ops->tick_getstatus);
+  if (lower->ops->tick_getstatus == NULL)
+    {
+      return -ENOTSUP;
```
Comment (raiden00pl) [inline include/nuttx/timers/timer.h:258]: missing
```
#include <errno.h>
```

**Comment by acassis** [comment]
@xiaoxiang781216 please include errno.h as Raiden commented

**Pull request closed without merging**